qualcommax: ipq60xx: add Linksys MR7500 support

PIPIPIG233666 opened the pull request:
Codename: Divo

Hardware specification:
========
SoC: Qualcomm IPQ6018
RAM: 512MB (2x ESMT 256MB DDR3L M15T2G16128A–DEBG2R)
NAND Flash: 512MB (Macronix MX30UF4G18AC or SK Hynix H27S4G8F2EDA-BC)
Ethernet: 1x 10/100/1000/2500/5000Mbps (Marvell AQR114C-B0)
Ethernet: 4x 10/100/1000Mbps (Qualcomm QCA8075)
WiFi1: 6GHz ax 4x4@20/40/160 MHz (Qualcomm QCN9024 + Skyworks SKY85784-11) - channels 33-229
WiFi2: 5GHz ax 2x2@20/40/80 MHz (Qualcomm QCN5052 + Skyworks SKY85755-11) - channels 36-177
WiFi3: 2.4GHz ax 2x2@20/40 MHz (Qualcomm QCN5022 + Skyworks SKY8340-11)
IoT: Bluetooth 5 (CSR8811) - not implemented
LED: 1x RGB status + USB Blue (PWM)
USB: 1x USB 3.0
Button: WPS, Reset

Flash instructions (Without Serial):
========
Open Linksys Web UI - http://192.168.1.1/ca or http://linksysxxxxx.lan/ca depending on your setup.
  xxxxx is the last 5 digits from the SN found on a sticker under the device.
Click on the Linksys Logo to by-pass smart app registration.
Login with your admin password. The default password can be found on the same sticker.
To enter into the support mode, click on the “CA” link and the bottom of the page.
Open the “Connectivity” menu and upload the squash-factory image with the “Choose file” button.
Click start. Ignore all the prompts and warnings by click “yes” in all the popups.

Flash instructions (With Serial):
========
1. Installation using serial connection from OEM firmware (default login: root, password: admin):
- fw_printenv -n boot_part
In case of 2:
- flash_erase /dev/mtd13 0 0
- nandwrite -p /dev/mtd13 openwrt-qualcommax-ipq60xx-linksys_mr7500-squashfs-factory.bin
or in case of 1:
- flash_erase /dev/mtd15 0 0
- nandwrite -p /dev/mtd15 openwrt-qualcommax-ipq60xx-linksys_mr7500-squashfs-factory.bin
After first boot install firmware on second partition:
- mtd -r -e kernel -n write openwrt-qualcommax-ipq60xx-linksys_mr7500-squashfs-factory.bin kernel
or:
- mtd -r -e alt_kernel -n write openwrt-qualcommax-ipq60xx-linksys_mr7500-squashfs-factory.bin alt_kernel

2. Installation from initramfs image using USB drive:
Put the initramfs image on the USB drive:
- dd bs=1M if=openwrt-qualcommax-ipq60xx-linksys_mr7500-initramfs-uImage.itb of=/dev/sda
Stop u-boot and run:
- usb start && usbboot $loadaddr 0 && bootm $loadaddr
Write firmware to the flash from initramfs:
- mtd -e kernel -n write openwrt-qualcommax-ipq60xx-linksys_mr7500-squashfs-factory.bin kernel
and:
- mtd -r -e alt_kernel -n write openwrt-qualcommax-ipq60xx-linksys_mr7500-squashfs-factory.bin alt_kernel

3. Back to the OEM firmware:
- mtd -e kernel -n write FW_MR7500_1.1.12.211919_prod.img kernel
and:
- mtd -r -e alt_kernel -n write FW_MR7500_1.1.12.211919_prod.img alt_kernel

4. USB recovery:
Put the initramfs image on the USB:
- dd bs=1M if=openwrt-qualcommax-ipq60xx-linksys_mr7500-initramfs-uImage.itb of=/dev/sda
Set u-boot env:
- fw_setenv bootusb 'usb start && usbboot $loadaddr 0 && bootm $loadaddr'
- fw_setenv bootcmd 'run bootusb; if test $auto_recovery = no; then bootipq; elif test $boot_part = 1; then run bootpart1; else run bootpart2; fi'

AQR firmware:
========
0. Firmware extracting:
To extract the firmware, use unblob (unblob.org)
1. Firmware loading:
To properly load the firmware and initialize AQR PHY, we must use the u-boot aq_load_fw function.
To do this, you need to modify u-boot env:
With USB recovery:
- fw_setenv bootcmd 'aq_load_fw; run bootusb; if test $auto_recovery = no; then bootipq; elif test $boot_part = 1; then run bootpart1; else run bootpart2; fi'
and without:
- fw_setenv bootcmd 'aq_load_fw; if test $auto_recovery = no; then bootipq; elif test $boot_part = 1; then run bootpart1; else run bootpart2; fi'
2. Firmware updating:
Newer firmware (AQR114C.cld) is available in the latest OEM firmware (https://downloads.linksys.com/support/assets/firmware/FW_MR7500_1.1.12.211919_prod.img).
Copy AQR114C.cld to /lib/firmware/marvell

Link: https://github.com/openwrt/openwrt/pull/17428

Signed-off-by: Weikai Kong <priv@pppig236.com>

PIPIPIG233666 committed:
qualcommax: ipq60xx: add Linksys MR7500 support

Codename: Divo

Hardware specification:
========
SoC: Qualcomm IPQ6018
RAM: 512MB (2x ESMT 256MB DDR3L M15T2G16128A–DEBG2R)
NAND Flash: 512MB (Macronix MX30UF4G18AC or SK Hynix H27S4G8F2EDA-BC)
Ethernet: 1x 10/100/1000/2500/5000Mbps (Marvell AQR114C-B0)
Ethernet: 4x 10/100/1000Mbps (Qualcomm QCA8075)
WiFi1: 6GHz ax 4x4@20/40/160 MHz (Qualcomm QCN9024 + Skyworks SKY85784-11) - channels 33-229
WiFi2: 5GHz ax 2x2@20/40/80 MHz (Qualcomm QCN5052 + Skyworks SKY85755-11) - channels 36-177
WiFi3: 2.4GHz ax 2x2@20/40 MHz (Qualcomm QCN5022 + Skyworks SKY8340-11)
IoT: Bluetooth 5 (CSR8811) - not implemented
LED: 1x RGB status + USB Blue (PWM)
USB: 1x USB 3.0
Button: WPS, Reset

Flash instructions (Without Serial):
========
Open Linksys Web UI - http://192.168.1.1/ca or http://linksysxxxxx.lan/ca depending on your setup.
  xxxxx is the last 5 digits from the SN found on a sticker under the device.
Click on the Linksys Logo to by-pass smart app registration.
Login with your admin password. The default password can be found on the same sticker.
To enter into the support mode, click on the “CA” link and the bottom of the page.
Open the “Connectivity” menu and upload the squash-factory image with the “Choose file” button.
Click start. Ignore all the prompts and warnings by click “yes” in all the popups.

Flash instructions (With Serial):
========
1. Installation using serial connection from OEM firmware (default login: root, password: admin):
- fw_printenv -n boot_part
In case of 2:
- flash_erase /dev/mtd13 0 0
- nandwrite -p /dev/mtd13 openwrt-qualcommax-ipq60xx-linksys_mr7500-squashfs-factory.bin
or in case of 1:
- flash_erase /dev/mtd15 0 0
- nandwrite -p /dev/mtd15 openwrt-qualcommax-ipq60xx-linksys_mr7500-squashfs-factory.bin
After first boot install firmware on second partition:
- mtd -r -e kernel -n write openwrt-qualcommax-ipq60xx-linksys_mr7500-squashfs-factory.bin kernel
or:
- mtd -r -e alt_kernel -n write openwrt-qualcommax-ipq60xx-linksys_mr7500-squashfs-factory.bin alt_kernel

2. Installation from initramfs image using USB drive:
Put the initramfs image on the USB drive:
- dd bs=1M if=openwrt-qualcommax-ipq60xx-linksys_mr7500-initramfs-uImage.itb of=/dev/sda
Stop u-boot and run:
- usb start && usbboot $loadaddr 0 && bootm $loadaddr
Write firmware to the flash from initramfs:
- mtd -e kernel -n write openwrt-qualcommax-ipq60xx-linksys_mr7500-squashfs-factory.bin kernel
and:
- mtd -r -e alt_kernel -n write openwrt-qualcommax-ipq60xx-linksys_mr7500-squashfs-factory.bin alt_kernel

3. Back to the OEM firmware:
- mtd -e kernel -n write FW_MR7500_1.1.12.211919_prod.img kernel
and:
- mtd -r -e alt_kernel -n write FW_MR7500_1.1.12.211919_prod.img alt_kernel

4. USB recovery:
Put the initramfs image on the USB:
- dd bs=1M if=openwrt-qualcommax-ipq60xx-linksys_mr7500-initramfs-uImage.itb of=/dev/sda
Set u-boot env:
- fw_setenv bootusb 'usb start && usbboot $loadaddr 0 && bootm $loadaddr'
- fw_setenv bootcmd 'run bootusb; if test $auto_recovery = no; then bootipq; elif test $boot_part = 1; then run bootpart1; else run bootpart2; fi'

AQR firmware:
========
0. Firmware extracting:
To extract the firmware, use unblob (unblob.org)
1. Firmware loading:
To properly load the firmware and initialize AQR PHY, we must use the u-boot aq_load_fw function.
To do this, you need to modify u-boot env:
With USB recovery:
- fw_setenv bootcmd 'aq_load_fw; run bootusb; if test $auto_recovery = no; then bootipq; elif test $boot_part = 1; then run bootpart1; else run bootpart2; fi'
and without:
- fw_setenv bootcmd 'aq_load_fw; if test $auto_recovery = no; then bootipq; elif test $boot_part = 1; then run bootpart1; else run bootpart2; fi'
2. Firmware updating:
Newer firmware (AQR114C.cld) is available in the latest OEM firmware (https://downloads.linksys.com/support/assets/firmware/FW_MR7500_1.1.12.211919_prod.img).
Copy AQR114C.cld to /lib/firmware/marvell

Link: openwrt#17428

Signed-off-by: Weikai Kong <priv@pppig236.com>

PIPIPIG233666 committed:
ipq6018-ess: Add dp5-syn node

Linksys MR7500 features 4x QSGMII QCA8075 and 1x USXGMII (1/2.5/5 GbE)

Signed-off-by: Weikai Kong <priv@pppig236.com>

PIPIPIG233666 committed:
ipq60xx: define common Linksys MR devices configuration

Signed-off-by: Weikai Kong <priv@pppig236.com>

PIPIPIG233666 committed:
qca-ssdk: enable AQR for ipq60xx targets

MR7500 features IPQ6018 which is CPPE, however, it also utilizes AQR114CB0 as WAN.
Previously, without this patch, aquantia_phy_api_ops was never called.
Therefore, enable `IN_AQUATIA_PHY` so that AQR PHY gets initialized properly on boot.

[    3.973942] Aquantia AQR114C 90000.mdio-1:08: loading firmware version 'v5.6.5 Cybertan Divo 090221 14:43:44' from 'NVMEM'
[   16.059953] aquantia_phy_api_ops_init[1485]:INFO:qca probe aquantia phy driver succeeded!
[   17.930149] Aquantia AQR114C 90000.mdio-1:08: attached PHY driver (mii_bus:phy_addr=90000.mdio-1:08, irq=POLL)

Signed-off-by: Weikai Kong <priv@pppig236.com>

RolandoMagico commented:
> Flash: 512MB (2x ESMT 256MB DDR3L M15T2G16128A–DEBG2R)

Shouldn’t this be RAM instead of FLASH?

robimarko commented:
Since you are flashing the AQR firmware to flash anyway, please use kernel NVMEM support for loading the firmware instead of having to rely on U-Boot, look at Qnap 301W.

robimarko commented on package/kernel/qca-ssdk/patches/104-enable-AQR-for-CPPE.patch:
```diff
@@ -0,0 +1,12 @@
+ 	IN_SFP=TRUE
+ 	IN_MALIBU_PHY=TRUE
+ else ifeq (CPPE, $(CHIP_TYPE))
++	IN_AQUANTIA_PHY=TRUE
```
Instead of patching SSDK more, you can just pass this via the OpenWrt SSDK makefile for ipq60xx target

robimarko commented on target/linux/qualcommax/image/ipq60xx.mk:
```diff
@@ -24,19 +24,25 @@ define Device/cambiumnetworks_xe3-4
-	IMAGE_SIZE := 75776k
+	KERNEL_SIZE := 8192k
+	NAND_SIZE := 512m
+	IMAGE_SIZE := 147456k
```
Please dont set these if you are going to have to override them later, they are not common then

robimarko commented on target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq6018-mr7500.dts:
```diff
@@ -0,0 +1,601 @@
+
+&tlmm {
+	pinctrl-0 = <&sd_ldo_pins>;
+	pinctrl-names = "default";
```
Ugh, what?

This is the provider node, not the consumer, drop all of the `pinctrl-0` and `pinctrl-names` from `tlmm` and call them from the actual users.

robimarko commented on target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq6018-mr7500.dts:
```diff
@@ -0,0 +1,601 @@
+			drive-strength = <8>;
+			bias-pull-down;
+		};
+	};
```
Newline after each subnode

robimarko commented on target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq6018-mr7500.dts:
```diff
@@ -0,0 +1,601 @@
+		};
+	};
+	hsuart_pins: hsuart_pins {
+		mux {
```
There is no need for a subnode in cases like this where you only have one function for all pins.

robimarko commented on target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq6018-mr7500.dts:
```diff
@@ -0,0 +1,601 @@
+			gpios = <&tlmm 56 GPIO_ACTIVE_LOW>;
+		};
+	};
+};
```
None of the stuff you put under `soc` belongs here, it all goes below `chosen`.

PIPIPIG233666 commented:
> Since you are flashing the AQR firmware to flash anyway, please use kernel NVMEM support for loading the firmware instead of having to rely on U-Boot, look at Qnap 301W.

Unfortunately, PHY is not initialized if aq_fw_load is not run in U-Boot.
See [MX8500 ](https://forum.openwrt.org/t/openwrt-support-for-linksys-mx8500/190742/182?u=pppig236)for more details.

PIPIPIG233666 commented:
> > LED: 1x RGB status (PWM) - TBD
> > USB: 1x USB 3.0 - TBD
> 
> What does TBD mean in this case?

To-be-done, PWM needs this [pending patch from upstream](https://[lore.kernel.org/lkml/20231005160550.2423075-2-quic_devipriy@quicinc.com/raw) and USB hasn't been fully working yet.

PIPIPIG233666 committed:
qualcommax: add PWM node to ipq6018 dts

Link: https://patches.linaro.org/project/linux-devicetree/cover/20231005160550.2423075-1-quic_devipriy@quicinc.com/

Signed-off-by: Weikai Kong <priv@pppig236.com>

PIPIPIG233666 committed:
qualcommax/ipq60xx: enable PWM

defconfig updated via `make kernel_menuconfig CONFIG_TARGET=subtarget`

Signed-off-by: Weikai Kong <priv@pppig236.com>

PIPIPIG233666 commented:
PWM IPQ patchset from https://patches.linaro.org/project/linux-devicetree/cover/20231005160550.2423075-1-quic_devipriy@quicinc.com/ , let me know if I need to add more info to the patch header.

On the other hand, no idea what the current status is, but V15 (the revision I included) is marked as superceded. I cannot find the newer one though.

robimarko commented:
> > Since you are flashing the AQR firmware to flash anyway, please use kernel NVMEM support for loading the firmware instead of having to rely on U-Boot, look at Qnap 301W.
> 
> Unfortunately, PHY is not initialized if aq_fw_load is not run in U-Boot. See [MX8500 ](https://forum.openwrt.org/t/openwrt-support-for-linksys-mx8500/190742/182?u=pppig236)for more details.

Then there is no point in defining NVMEM support for loading the FW in DTS at all

robimarko commented on package/kernel/qca-ssdk/Makefile:
```diff
@@ -58,6 +58,7 @@ endif
 
 ifeq ($(CONFIG_TARGET_SUBTARGET), "ipq60xx")
     MAKE_FLAGS+= CHIP_TYPE=CPPE
```
```suggestion
    MAKE_FLAGS+= CHIP_TYPE=CPPE \
```

robimarko commented on package/kernel/qca-ssdk/Makefile:
```diff
@@ -58,6 +58,7 @@ endif
 
 ifeq ($(CONFIG_TARGET_SUBTARGET), "ipq60xx")
     MAKE_FLAGS+= CHIP_TYPE=CPPE
+    MAKE_FLAGS+= IN_AQUANTIA_PHY=TRUE
```
```suggestion
    IN_AQUANTIA_PHY=TRUE
```

Then there is no need for `MAKE_FLAGS` again

PIPIPIG233666 commented:
BT is up but not working properly due to the requirements of modifying PSR, should I even try to fix it?

robimarko commented on target/linux/qualcommax/ipq60xx/config-default:
Just merge this one with the commit adding PWM support.

BTW, has this kernel config been updated via kernel_menuconfig or by hand?

robimarko commented on target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq6018-ess.dtsi:
```diff
@@ -504,4 +504,15 @@
+		reg = <0x0 0x3a003000 0x0 0x3fff>;
+		qcom,mactype = <1>;
+		local-mac-address = [000000000000];
+		phy-mode = "psgmii";
```
This should be `usxgmii` by default

robimarko commented on target/linux/qualcommax/ipq60xx/base-files/etc/board.d/01_leds:
```diff
@@ -19,6 +19,18 @@ linksys,mr7350)
+	ucidef_set_led_netdev "lan4-port-link" "LAN4-PORT-LINK" "90000.mdio-1:03:green:lan" "lan4" "link_10 link_100 link_1000"
+	ucidef_set_led_netdev "lan4-port-traffic" "LAN4-PORT-TRAFFIC" "90000.mdio-1:03:orange:lan" "lan4" "tx rx link_10 link_100 link_1000"
+	ucidef_set_led_netdev "wan-port-link" "WAN-PORT-LINK" "90000.mdio-1:08:green:lan" "wan" "link_10 link_100 link_1000 link_2500 link_5000"
+	ucidef_set_led_netdev "wan-port-traffic" "WAN-PORT-TRAFFIC" "90000.mdio-1:08:orange:wan" "wan" "tx rx link_10 link_100 link_1000 link_2500 link_5000"
```
Are you sure that vendor AQR FW doesn't already pre-provision the LED registers?

PIPIPIG233666 replied:
It doesn't look like the WAN LEDs work without this

PIPIPIG233666 commented:
> PWM driver was included as part of ipq50xx, please rebase

Done

robimarko commented on target/linux/qualcommax/ipq60xx/config-default:
```diff
@@ -9,3 +12,7 @@ CONFIG_REGULATOR_CPR3=y
 # CONFIG_REGULATOR_CPR3_NPU is not set
 CONFIG_REGULATOR_CPR4_APSS=y
 CONFIG_REGULATOR_QCOM_SMD_RPM=y
+CONFIG_SKB_RECYCLER=y
```
This does not look like an mainline option at all

robimarko commented on target/linux/qualcommax/ipq60xx/config-default:
This is not a refresh since you are enabling PWM, just make the title and description match what you are really doing

robimarko commented on package/boot/uboot-envtools/files/qualcommax_ipq60xx:
This was simplified, so please rebase on top of master

robimarko commented on target/linux/qualcommax/image/ipq60xx.mk:
```diff
@@ -45,6 +45,17 @@ define Device/linksys_mr7350
+	NAND_SIZE := 512m
+	IMAGE_SIZE := 147456k
+	DEVICE_PACKAGES := ipq-wifi-linksys_mr7500 \
+		ath11k-firmware-qcn9074 kmod-ath11k-pci kmod-bluetooth
```
Is there a point in including Bluetooth kmod since it most likely does not work at all

robimarko commented on target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq6018-mr7500.dts:
```diff
@@ -0,0 +1,648 @@
+	model = "Linksys MR7500";
+	compatible = "linksys,mr7500", "qcom,ipq6018";
+	#address-cells = <0x2>;
+	#size-cells = <0x2>;
```
Both address and size cells are already set in the SoC DTSI

robimarko commented on target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq6018-mr7500.dts:
```diff
@@ -0,0 +1,648 @@
+	};
+	
+	leds {
+		compatible = "pwm-leds";
```
Missing newline

robimarko commented on target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq6018-mr7500.dts:
```diff
@@ -0,0 +1,648 @@
+	leds {
+		compatible = "pwm-leds";
+		usb {
+			label = "usb:system";
```
Please no labels, just use function and color instead.

Same goes for all other LED-s

PIPIPIG233666 replied:
looks like the leds broke after getting rid of the label and adding color & function, nothing is in the log, am i missing something?

PIPIPIG233666 replied:
nvm, i was missing some drivers

robimarko commented on target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq6018-mr7500.dts:
```diff
@@ -0,0 +1,648 @@
+	};
+	
+	output-usb0-power {
+		compatible = "regulator-output";
```
No need for this

robimarko replied:
Just pass the `reg_usb_vbus` to qusb PHY via:
`vdd-supply = <&reg_usb_vbus>;`

robimarko commented on target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq6018-mr7500.dts:
```diff
@@ -0,0 +1,648 @@
+		regulator-min-microvolt = <5000000>;
+		regulator-max-microvolt = <5000000>;
+		gpio = <&tlmm 25 GPIO_ACTIVE_LOW>;
+		regulator-always-on;
```
Yeah, I doubt this is required, maybe `regulator-boot-on`

PIPIPIG233666 replied:
I think I did this for usb to work, otherwise it is detected but immediately rejects the thumb drive for some reason

PIPIPIG233666 replied:
nvm, it works now with only boot on

robimarko commented on target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq6018-mr7500.dts:
```diff
@@ -0,0 +1,648 @@
+		regulator-max-microvolt = <5000000>;
+		gpio = <&tlmm 25 GPIO_ACTIVE_LOW>;
+		regulator-always-on;
+		/* enable-active-low; */
```
Remove it

robimarko commented on target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq6018-mr7500.dts:
```diff
@@ -0,0 +1,648 @@
+			drive-strength = <8>;
+			bias-pull-down;
+		};
+	};
```
Always add newlines

robimarko commented on target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq6018-mr7500.dts:
```diff
@@ -0,0 +1,648 @@
+		};
+	};
+	uart_pins: uart_pins {
+		mux {
```
There is no point in having a subnode here, just put it directly under `uart_pins`

robimarko commented on target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq6018-mr7500.dts:
```diff
@@ -0,0 +1,648 @@
+		};
+	};
+	sd_ldo_pins: sd_ldo_pins {
+		pinctrl-0 = <&sd_ldo_pins>;
```
This is obviously invalid, you cannot be calling pinctrl on itself nor it makes any sense, call it under the that requires it

robimarko commented on target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq6018-mr7500.dts:
```diff
@@ -0,0 +1,648 @@
+	sd_ldo_pins: sd_ldo_pins {
+		pinctrl-0 = <&sd_ldo_pins>;
+		pinctrl-names = "default";
+		mux {
```
Again, no need for subnodes when its just single function, same goes all others

robimarko commented on target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq6018-mr7500.dts:
```diff
@@ -0,0 +1,648 @@
+		qca8075_0: ethernet-phy@0 {
+			compatible = "ethernet-phy-ieee802.3-c22";
+			reg = <0>;
+			leds {
```
Again, newlines, always newlines between, it goes universally for the whole DTS

robimarko commented on target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq6018-mr7500.dts:
```diff
@@ -0,0 +1,648 @@
+	};
+	aqr114c: ethernet-phy@8 {
+		reg = <8>;
+		reset-gpios = <&tlmm 77 GPIO_ACTIVE_HIGH>;
```
If my memory is correct, AQR-s always use ACTIVE LOW resets

PIPIPIG233666 replied:
This is from stock tho:
https://github.com/PIPIPIG233666/openwrt-legacy/blob/0848d76c2fc254be5c46e9b19a22e70abac3236d/arch/arm64/boot/dts/qcom/qcom-ipq6018-cp01.dtsi#L277

> 		phy-reset-gpio = <&tlmm 75 0 &tlmm 77 1>;

PIPIPIG233666 commented on target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq6018-mr7500.dts:
```diff
@@ -0,0 +1,648 @@
+		#size-cells = <1>;
+		
+		partitions {
+			compatible = "fixed-partitions";
```
good question, now i removed the dts partitions and im stuck with 
```[    1.216965] UBI error: cannot open mtd rootfs, er�[    1.228399] Waiting for root device /dev/ubiblock0_0...
```
any insight to this? do i need t odo some advanced formatting to bring the old layout back?

PIPIPIG233666 replied:
actually now i remember: back then was attempting to use nvmem for the aqr fw loading

PIPIPIG233666 replied:
done

robimarko commented on target/linux/qualcommax/patches-6.6/100-arm64-dts-qcom-ipq6018-add-pwm-node.patch:
```diff
@@ -0,0 +1,25 @@
+-			compatible = "qcom,tcsr-ipq6018", "syscon";
++			compatible = "qcom,tcsr-ipq6018", "syscon", "simple-mfd";
+ 			reg = <0x0 0x01937000 0x0 0x21000>;
++			ranges = <0x0 0x0 0x01937000 0x21000>;
```
Hm, this looks incorrect to me

robimarko replied:
Though it seems to be correct:
https://patches.linaro.org/project/linux-devicetree/patch/20231005043127.2690639-5-quic_devipriy@quicinc.com/

Use the latest patch please, and keep it whole with the full header so git am can be used to apply it and we actually know what its about

robimarko commented on target/linux/qualcommax/ipq60xx/base-files/etc/board.d/02_network:
```diff
@@ -49,6 +50,11 @@ ipq60xx_setup_macs()
 		lan_mac=$label_mac
 		wan_mac=$label_mac
 		;;
+	linksys,mr7500)
```
Use the ascii-env NVMEM driver instead, example:
https://github.com/openwrt/openwrt/commit/eacc4d8c9b31ac17df034d2140558bdaaa56c16b

PIPIPIG233666 replied:
can i do it without touching the smem?

PIPIPIG233666 replied:
ok, i tried it and it seems to work; however, bootlog complains about bad cell count:
```
Thu Mar  6 08:10:29 2025 kern.notice kernel: [    1.221988] 20 qcomsmem partitions found on MTD device qcom_nand.0
Thu Mar  6 08:10:29 2025 kern.notice kernel: [    1.229058] Creating 20 MTD partitions on "qcom_nand.0":
Thu Mar  6 08:10:29 2025 kern.notice kernel: [    1.235315] 0x000000000000-0x000000180000 : "0:sbl1"
Thu Mar  6 08:10:29 2025 kern.notice kernel: [    1.242742] 0x000000180000-0x000000280000 : "0:mibib"
Thu Mar  6 08:10:29 2025 kern.notice kernel: [    1.247116] 0x000000280000-0x000000600000 : "0:qsee"
Thu Mar  6 08:10:29 2025 kern.notice kernel: [    1.254405] 0x000000600000-0x000000680000 : "0:devcfg"
Thu Mar  6 08:10:29 2025 kern.notice kernel: [    1.256645] 0x000000680000-0x000000700000 : "0:rpm"
Thu Mar  6 08:10:29 2025 kern.notice kernel: [    1.261721] 0x000000700000-0x000000780000 : "0:cdt"
Thu Mar  6 08:10:29 2025 kern.notice kernel: [    1.266418] 0x000000780000-0x000000800000 : "0:appsblenv"
Thu Mar  6 08:10:29 2025 kern.notice kernel: [    1.271326] 0x000000800000-0x000000980000 : "0:appsbl"
Thu Mar  6 08:10:29 2025 kern.notice kernel: [    1.277723] 0x000000980000-0x000000a00000 : "0:art"
Thu Mar  6 08:10:29 2025 kern.notice kernel: [    1.281985] 0x000000a00000-0x000000a80000 : "0:ethphyfw"
Thu Mar  6 08:10:29 2025 kern.notice kernel: [    1.286639] 0x000000a80000-0x000000b00000 : "u_env"
Thu Mar  6 08:10:29 2025 kern.notice kernel: [    1.292203] 0x000000b00000-0x000000b40000 : "s_env"
Thu Mar  6 08:10:29 2025 kern.notice kernel: [    1.296538] 0x000000b40000-0x000000b80000 : "devinfo"
Thu Mar  6 08:10:29 2025 kern.notice kernel: [    1.301511] 1 fixed-partitions partitions found on MTD device devinfo
Thu Mar  6 08:10:29 2025 kern.notice kernel: [    1.305877] Creating 1 MTD partitions on "devinfo":
Thu Mar  6 08:10:29 2025 kern.notice kernel: [    1.312316] 0x000000000000-0x000000040000 : "env-data"
Thu Mar  6 08:10:29 2025 kern.err kernel: [    1.317480] OF: Bad cell count for /soc@0/nand-controller@79b0000/nand@0/partitions/partition-0-devinfo
Thu Mar  6 08:10:29 2025 kern.notice kernel: [    1.363153] 0x000000b80000-0x000001380000 : "kernel"
Thu Mar  6 08:10:29 2025 kern.notice kernel: [    1.371233] 0x000001380000-0x000005d80000 : "rootfs"
Thu Mar  6 08:10:29 2025 kern.info kernel: [    1.440468] mtd: setting mtd15 (rootfs) as root device
Thu Mar  6 08:10:29 2025 kern.alert kernel: [    1.440830] mtdsplit: no squashfs found in "rootfs"
Thu Mar  6 08:10:29 2025 kern.notice kernel: [    1.444555] 0x000005d80000-0x000006580000 : "alt_kernel"
Thu Mar  6 08:10:29 2025 kern.notice kernel: [    1.457221] 0x000006580000-0x00000af80000 : "alt_rootfs"
Thu Mar  6 08:10:29 2025 kern.notice kernel: [    1.526649] 0x00000af80000-0x00000b380000 : "sysdiag"
Thu Mar  6 08:10:29 2025 kern.notice kernel: [    1.530963] 0x00000b380000-0x00001f080000 : "syscfg"
Thu Mar  6 08:10:29 2025 kern.notice kernel: [    1.825929] 0x00001f080000-0x00001f580000 : "app_data"
```

PIPIPIG233666 replied:
ok fixed the warning, was missing an addr/size 
```
                partitions {
                        compatible = "qcom,smem-part";
+                       #address-cells = <1>;
+                       #size-cells = <1>;
 
                        partition-0-devinfo {
                                compatible = "fixed-partitions";
@@ -477,7 +479,6 @@
                                        reg = <0x0 0x40000>;
 
                                        nvmem-layout {
-
                                                compatible = "ascii-eq-delim-env";
                                                #address-cells = <1>;
                                                #size-cells = <1>;

```

robimarko commented on target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq6018-mr7500.dts:
```diff
@@ -0,0 +1,517 @@
+		ethernet1 = &dp2;
+		ethernet2 = &dp3;
+		ethernet3 = &dp4;
+		ethernet4 = &dp5_syn;
```
If you later need to fixup these then there is no point in even having the aliases

PIPIPIG233666 commented on target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq6018-mr7500.dts:
```diff
@@ -0,0 +1,551 @@
+		#size-cells = <2>;
+		ranges;
+		
+		wifi {
```
> wifi
does it even matter if i dont add wifi@1,0 here? seems like theres no functional differences
